polkadot/node/core/prospective-parachains/src/metrics.rs

```diff
@@ -46,6 +78,34 @@ impl metrics::Metrics for Metrics {
 				))?,
 				registry,
 			)?,
+			introduce_seconded_candidate: prometheus::register(
+				prometheus::Histogram::with_opts(prometheus::HistogramOpts::new(
+					"polkadot_parachain_prospective_parachains_introduce_seconded_candidate",
+					"Time spent within `prospective_parachains::handle_introduce_seconded_candidate`",
+				))?,
+				registry,
+			)?,
+			hypothetical_membership: prometheus::register(
+				prometheus::Histogram::with_opts(prometheus::HistogramOpts::new(
+					"polkadot_parachain_prospective_parachains_hypothetical_membership",
+					"Time spent responding to `GetHypotheticalMembership`",
+				))?,
+				registry,
+			)?,
+			candidate_storage_count: prometheus::register(
+				Gauge::new(
+					"polkadot_parachain_prospective_parachains_candidate_storage_count",
+					"Total size of the prospective parachains candidate storage".to_string(),
+				)?,
+				registry,
+			)?,
+			candidate_storage_unconnected_count: prometheus::register(
+				Gauge::new(
+					"polkadot_parachain_prospective_parachains_candidate_storage_unconnected_count",
+					"Number of unconnected candidates in the prospective parachains candidate storage".to_string(),
+				)?,
+				registry,
+			)?,
```

Comment:
I would define a single metric and have labels name `kind` with values `connected`, `unconnected`.